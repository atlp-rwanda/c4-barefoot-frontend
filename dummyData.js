export const user = [
  {
    "first_name": "M",
    "last_name": "JAckson",
    "email": "jackson.novelty@gmail.com",
    "occupation": "Programming",
    "username": "jacks",
    "bio": "Software Developer",
    "address": "kk 509 st",
    "language": "Kinyarwanda",
    "password": "12345678",
    "profile_picture": "https://res.cloudinary.com/mjackson/image/upload/v1609304524/Screen_Shot_2020-12-29_at_08.21.43_qe7pem.png"
  }
]

export const userProfile = {
  address: "Kigali",
  bio: null,
  email: "manager_id@gmail.com",
  first_name: "Mugabo",
  id: "38eb202c - 3f67 - 4eed - b7ac - 9c31bc226e0c",
  language: "French",
  last_name: "Manager",
  manager_id: null,
  occupation: "Software Developer",
  profile_picture: "https://res.cloudinary.com/mjackson/image/upload/v1611135351/Albert-einstein-profile-picture-512x512.png.cf_ouqza2.png",
  refreshtoken: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiNzI1NGE5ZTctMmUxYi00ZjgzLWFkNzMtNzhiOTBkZDNkZjc3IiwidXNlcm5hbWUiOiJtYW5hZ2VyX2lkIiwiaWF0IjoxNjExMTMzNDIwLCJleHAiOjE2MTE3MzgyMjB9.2X_RFNNdIGr-hc1_h63yBuLMGNsOopt2cRAW4jXT4fc",
  user_role_id: "7254a9e7-2e1b-4f83-ad73-78b90dd3df77",
  username: "manager_id",
  verified: true
}

export const signupState = {
    requesting: false,
    errorOpen: false,
    error: ""
  }
export const locationsPayload = {
  rows:[
    {
      id: "c6028e0d-ef88-4693-ab49-f37669891724",
      LocationName: "Kigali",
      country: "Rwanda",
      description: "The land of a thousand hills",
      link: "https://www.nationalgeographic.com/content/dam/travel/Guide-Pages/africa/downtown-kigali-rwanda.adapt.1900.1.jpg",
    }
  ]}

export const accommodationsPayload = {
  rows:[
    {
      
        id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        country: "Rwanda",
        city: "Kigali",
        title: "Marriot Hotel",
        description: "A serene environment for relaxation",
        photos: "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
    
    }
  ]}

  export const accommodationAminitiesPayload ={
    "singleAccommodation": {
        "id": "7a7ce9e2-ce8a-4d76-9626-cfabf2009e42",
        "country": "Rwanda",
        "city": "Musanze",
        "state": "Rusizi",
        "streetAddress": "KN 22 ST",
        "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
        "propertyType": "Hotel",
        "numberOfRooms": 500,
        "price": 3000000,
        "typeOfBed": "Decker",
        "title": "bisate lodge",
        "description": "Bisate is located in the natural amphitheatre of an eroded volcanic cone, with dramatic views of the peaks of the Bisoke and Karisimbi volcanoes rearing up through Afro-alpine forests.",
        "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616576198/bisate-lodge-9_qjbmmf.jpg"
    },
    "amenities": {
        "wifi": false,
        "airConditioner": true,
        "shampoo": false,
        "ironing": true,
        "tv": false,
        "smokeDetector": true,
        "fireExtinguisher": false,
        "lockOnDoor": true
    }
}
  export const rolePayload = {
    rows:[
      {
        
          id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
          name: "Requester",
          description: "requster"
      
      }
    ]}
  
    export const permissionsPayload = {
      permissions:{
        "edit profile": 0,"assign requesters to manager": 0,"create travel requests": 0,"view travel requests": 0,"edit travel requests": 0,"cancel travel requests": 0,"approve direct reports travel requests": 0,"view direct reports travel requests": 0,"reject direct reports travel requests": 0,"view accommodations": 0,"create accommodations": 0,"update accommodations": 0,"delete accommodations": 0,"book accommodations": 0,"view locations": 0,"create locations": 0,"update locations": 0,"delete locations": 0,"view statistics": 0
      }
  }

  export const allTravelPayload =[
    {
      travelRequestInfo: {
          travelId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
          managerId: "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
          userId: "83b2a3e7-9ba4-4d3f-b3a3-d31940ee2edc",
          status: "approved",
          createdAt: "2021-01-11T20:08:41.421Z",
          updatedAt: "2021-03-01T08:34:29.612Z",
          Trip: [
              {
                  tripId: "83b2a3e7-9ba4-3f4d-b3a3-d31940ee2edc",
                  originCity: "Kigali",
                  destination: "Cairo",
                  reason: "Trippin",
                  tripDate: "2020-10-10T00:00:00.000Z",
                  returnDate: "2021-10-10T00:00:00.000Z",
                  accommodationId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
                  createdAt: "2021-01-11T20:08:41.427Z",
                  updatedAt: "2021-01-11T20:08:41.427Z",
                  travelId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5"
              },
              {
                  tripId: "83b2a3e7-4ab9-3f4d-b3a3-d31940ee2edc",
                  originCity: "Kigali",
                  destination: "Kampala",
                  reason: "Trippin",
                  tripDate: "2020-10-10T00:00:00.000Z",
                  returnDate: "2021-10-10T00:00:00.000Z",
                  accommodationId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
                  createdAt: "2021-01-11T20:08:41.427Z",
                  updatedAt: "2021-01-11T20:08:41.427Z",
                  travelId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5"
              },
              {
                  tripId: "1a52f79e-568a-45b3-9151-4dfa40bb1217",
                  originCity:" Kigali",
                  destination: "Nairobi",
                  reason: "Trippin",
                  tripDate: "2020-10-10T00:00:00.000Z",
                  returnDate: "2021-10-10T00:00:00.000Z",
                  accommodationId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc8",
                  createdAt: "2021-01-11T20:08:41.427Z",
                  updatedAt: "2021-01-11T20:08:41.427Z",
                  travelId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5"
              },
              {
                  tripId: "aeebcf33-d125-44ce-b8c1-f5d5e8b75f13",
                  originCity: "Kigali",
                  destination: "Nairobi",
                  reason: "Trippin",
                  tripDate: "2020-10-10T00:00:00.000Z",
                  returnDate: "2021-10-10T00:00:00.000Z",
                  accommodationId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc8",
                  createdAt: "2021-01-11T20:08:41.427Z",
                  updatedAt: "2021-01-11T20:08:41.427Z",
                  travelId: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5"
              }
          ]
      },
      userInfo: {
          username: "requesterOne",
          firstName: "Requester",
          lastName: "One",
          profilePpicture: "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg"
      },
      accommodationInfo: [
          {
              id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
              country: "Rwanda",
              city: "Kigali",
              state: "Nyarugenge",
              streetAddress: "KN 22 ST",
              locationID: "c6028e0d-ef88-4693-ab49-f37669891724",
              propertyType: "Hostel",
              numberOfRooms: 100,
              typeOfBed: "Double Decker",
              title: "Kigali Hostels",
              description: "A serene environment for relaxation",
              photos: "image.png"
          },
          {
              id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
              country: "Rwanda",
              city: "Kigali",
              state: "Nyarugenge",
              streetAddress: "KN 22 ST",
              locationID: "c6028e0d-ef88-4693-ab49-f37669891724",
              propertyType: "Hostel",
              numberOfRooms: 100,
              typeOfBed: "Double Decker",
              title: "Kigali Hostels",
              description: "A serene environment for relaxation",
              photos: "image.png"
          },
          {
              error: "No info on this accomodation"
          },
          {
              error: "No info on this accomodation"
          }
      ]
  },
  {
      travelRequestInfo: {
          travelId: "6a631198-cb98-4925-82fc-b00be0d34e32",
          managerId: "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
          userId: "d74fcc5e-5755-4366-83ef-cf306b013c46",
          status: "rejected",
          createdAt: "2021-01-11T20:09:02.833Z",
          updatedAt: "2021-03-01T08:54:54.237Z",
          Trip: []
      },
      userInfo: {
          username: "pushrequester",
          firstName: "pushRequester",
          lastName: "One",
          profilePpicture: "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg"
      }
  },
  ]

  export const allTravelRequestState = {
    loading: false,
    travel: allTravelPayload,
    error: null
  }


  export const singleTravelPayload = [{
    travelRequestInfo: {
        travelId: "6a631198-cb98-4925-82fc-b00be0d34e32",
        managerId: "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
        userId: "d74fcc5e-5755-4366-83ef-cf306b013c46",
        status: "rejected",
        createdAt: "2021-01-11T20:09:02.833Z",
        updatedAt: "2021-03-01T08:54:54.237Z",
        Trip: []
    },
    userInfo: {
        username: "pushrequester",
        firstName: "pushRequester",
        lastName: "One",
        profilePpicture: "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg"
    }
}]
export const singleTravelState= {
  loading: false,
  error: null,
  travel: singleTravelPayload,
}

export const updateTravelState= {
  loading: false,
  error: null,
  travel: null
}

export const UserInfo= {
    username: "pushrequester",
    firstName: "pushRequester",
    lastName: "One",
    profilePpicture: "https://www.cobdoglaps.sa.edu.au/wp-content/uploads/2017/11/placeholder-profile-sq.jpg"  
}

export const  accommodationInfo= [
  {
      id: "598d90a3-68e6-43ac-970a-2855657be478",
      country: "Rwanda",
      city: "Kigali",
      state: "Kicukiro",
      streetAddress: "KN 45 ST",
      locationID: "c6028e0d-ef88-4693-ab49-f37669891724",
      propertyType: "Hostel",
      numberOfRooms: 100,
      typeOfBed: "Double Decker",
      title: "Marina Hotel",
      description: "A serene environment for relaxation",
      photos: "image2.png"
  }
]

export const  travelRequestInfo= {
  travelId: "59ba77ce-3649-4b8a-bed6-b127cc1c530f",
  managerId: "0ce36391-2c08-4703-bddb-a4ea8cccbbc5",
  userId: "d74fcc5e-5755-4366-83ef-cf306b013c46",
  status: "approved",
  createdAt: "2021-03-02T11:18:36.501Z",
  updatedAt: "2021-03-02T11:21:00.554Z",
  Trip: [
      {
          tripId: "98624254-da4c-47bd-91da-53d9d89172cc",
          originCity: "MUSANZE",
          destination: "KARONGI",
          reason: "Research and market development trip in rular areas",
          tripDate: "2020-12-20T08:53:31.303Z",
          returnDate: null,
          accommodationId: "598d90a3-68e6-43ac-970a-2855657be478",
          createdAt: "2021-03-02T11:18:36.600Z",
          updatedAt: "2021-03-02T11:18:36.600Z",
          travelId: "59ba77ce-3649-4b8a-bed6-b127cc1c530f"
      }
  ]}
export const accommodationState = {
  pending: true,
  accommodations: accommodationsPayload,
  accommodation: {},
  nation:null,
  amenities:{},
  accommodationsByLocation:[],
  selectedAccommodation:null,
  count:null,
  error: null,
  temp:null,
  accId:null
}

export const fetchAccommodationByLocationPayload={
  "status": 200,
  "page": 1,
  "accommodations": {
      "count": 15,
      "rows": [
          {
              "id": "122a0d86-8b78-4bb8-b28f-8e5f7811c456",
              "country": "Rwanda",
              "city": "Kigali",
              "state": "Kagarama",
              "streetAddress": "KN 22 ST",
              "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
              "propertyType": "Hotel",
              "numberOfRooms": 500,
              "price": 25000,
              "typeOfBed": "Decker",
              "title": "Garama hotel",
              "description": "Best bedrooms",
              "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616493369/93989612_kocs6i.jpg",
              "createdAt": "2021-03-24T10:33:51.926Z",
              "updatedAt": "2021-03-24T10:33:51.926Z"
          },
          {
              "id": "28debeec-6adc-4dbe-a3c0-d3480c49ee76",
              "country": "Rwanda",
              "city": "Nyungwe",
              "state": "Rusizi",
              "streetAddress": "KN 22 ST",
              "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
              "propertyType": "Hotel",
              "numberOfRooms": 500,
              "price": 500000,
              "typeOfBed": "Decker",
              "title": "Nyungwe hotel",
              "description": "Best bedrooms",
              "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616493171/Facts-about-Virunga-National-Park-750x450_cr4alx.jpg",
              "createdAt": "2021-03-24T10:33:51.926Z",
              "updatedAt": "2021-03-24T10:33:51.926Z"
          },
          {
              "id": "2d647115-3af7-4df0-99aa-6656c764829f",
              "country": "Rwanda",
              "city": "Kigali",
              "state": "Kagarama",
              "streetAddress": "KN 22 ST",
              "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
              "propertyType": "Hotel",
              "numberOfRooms": 500,
              "price": 5000000,
              "typeOfBed": "Double and single",
              "title": "One and Only",
              "description": "The best luxury hotel in the middle of the virunga national park in Rwanda",
              "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616572403/One-and-Only-Nyungwe-Huose-750x450_vgs415.jpg",
              "createdAt": "2021-03-24T10:33:51.926Z",
              "updatedAt": "2021-03-24T10:33:51.926Z"
          },
          {
              "id": "4d5b1b29-51e4-4db5-8b99-57fd76bd5544",
              "country": "Rwanda",
              "city": "Kigali",
              "state": "Kagarama",
              "streetAddress": "KN 22 ST",
              "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
              "propertyType": "Hotel",
              "numberOfRooms": 500,
              "price": 30000,
              "typeOfBed": "Decker",
              "title": "Hill Top hotel hotel",
              "description": "Best views",
              "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616493292/hotel-hilltop-country_fwr16e.jpg",
              "createdAt": "2021-03-24T10:33:51.926Z",
              "updatedAt": "2021-03-24T10:33:51.926Z"
          },
          {
              "id": "2830496d-65e3-4289-b6e9-819933637ca6",
              "country": "Rwanda",
              "city": "Kigali",
              "state": "Kiyovu",
              "streetAddress": "KN 22 ST",
              "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
              "propertyType": "Hotel",
              "numberOfRooms": 500,
              "price": 500000,
              "typeOfBed": "Decker",
              "title": "Marriot Hotel",
              "description": "Located in the beautiful and silent part of kigali Kiyovu",
              "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616493334/93100479_ovlmlf.jpg",
              "createdAt": "2021-03-24T10:33:51.926Z",
              "updatedAt": "2021-03-24T10:33:51.926Z"
          },
          {
              "id": "5efb6fd8-d43a-41b5-b021-d6fe1536b88b",
              "country": "Rwanda",
              "city": "Rubavu",
              "state": "Rubavu",
              "streetAddress": "KN 22 ST",
              "locationID": "23fd8031-52d6-4150-958c-c162ba9e8f5c",
              "propertyType": "Hotel",
              "numberOfRooms": 500,
              "price": 50000,
              "typeOfBed": "Decker",
              "title": "The Manor Hotel",
              "description": "The hotel at the beautiful Kivu beach, best place to spend your summer vaccation ",
              "photos": "https://res.cloudinary.com/barefootnomadcohort4/image/upload/v1616493273/themanorhotelsmall1_maevju.jpg",
              "createdAt": "2021-03-24T10:33:51.926Z",
              "updatedAt": "2021-03-24T10:33:51.926Z"
          }
      ]
  },
  "nation": "Rwanda"
}
export const locationState = {
  pending: false,
  locations: locationsPayload,
  error: null
}

export const bookAccommodationState={
  pending: false,
  snackBarMessage: {
    open: false,
    severity: '',
    message: null
  }
}

let a = 'https://barefoot-nomad-app-v1.herokuapp.com'
export const props = {
  CheckReturningAction : jest.fn(),
  checkTravelDatesAction: jest.fn(),
  getLocationsAction: jest.fn(),
  searchCurrentLocationAction: jest.fn(),
  selectAccommodationAction: jest.fn(),
  handleErrorsAction:jest.fn(),
  closeSnackbar: jest.fn(),
  addTravelReasonAction: jest.fn(),
  sendTravelRequestAction: jest.fn(),
  addMultiCityAction: jest.fn(),
  removeMultiCityAction: jest.fn(),
  openModalAction: jest.fn(),
}
export const props2={
  searchLocations: [],
  searchLocationsLoading: false,
  currentLocation: '',
  destinationLocation: '',
  isReturning: false,
  departureDate: '',
  returnDate: '',
  searchAccommodations: [],
  searchAccommodationsLoading:false,
  selectedAccommodation:[],
  selectedLocations:[],
  displaySelection:false,
  displaySelected: false,
  snackBarMessage:{
      open: false,
      message: null,
      severity: ''
  },
  success: false,
  sendLoading:false,
  travelReason: '',
  Modal: {
      open: false,
      data:{}
  }
}