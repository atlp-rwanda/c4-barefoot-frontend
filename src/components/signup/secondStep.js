import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Formik, Form, Field } from 'formik';
import { TextField, Select, InputAdornment, CircularProgress, MenuItem, FormControl, InputLabel } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import Box from '@material-ui/core/Box';
import { AccountCircle, Language, Subject, EventBusy } from "@material-ui/icons";
import ListItemIcon from '@material-ui/core/ListItemIcon';
import axios from 'axios'
import Avatar from '@material-ui/core/Avatar';
import { useTranslation } from 'react-i18next';


const useStyles = makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    '@media(max-width: 320px)' : {
      display: 'grid'
    },
  },
  button: {
    margin: theme.spacing(1)
  },
  large: {
    width: theme.spacing(7),
    height: theme.spacing(7),
  },
  button: {
    margin: theme.spacing(1),
  },
  label: {
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#d5d5d5',
    color: 'black',
    padding: '0.3rem',
    fontFamily: 'sans-serif',
    borderRadius: '0.3rem',
    cursor: 'pointer',
    margin: '1rem',
  },
  uploadSpinner: {
    width: '20px !important',
    height: '20px !important',
    margin: '0 5px -5px 5px'
  }
}));

export  const SecondStep = ({
  formData,
  setFormData,
  nextStep,
  prevStep
}) => {
  const { t, i18n } = useTranslation();
  const classes = useStyles();
  const [direction, setDirection] = useState('back');
  const [photo, setPhoto] = useState('');
  const [loading, setLoading] = useState(false);
  useEffect(() => { setLoading(false); setPhoto('') }, [])
  return (
    <>
      <Formik
        initialValues={formData}
        onSubmit={values => {
          setFormData(values);
          direction === 'back' ? prevStep() : nextStep();
        }}
      >
        {({ errors, touched, setFieldValue }) => (
          <Form className={classes.form}>
            <Box display='inline-flex' justifyContent='center' alignItems='center'>
              <Avatar 
              alt="user" 
              name='profile_picture' 
              margin='normal'
              src={photo ? photo : formData.profile_picture }
              className={classes.large}/>
              <input 
              id="file" 
              name="profile_picture" 
              type="file" 
              hidden={true}
              onChange={(e) => {
                const profile_picture = e.target.files[0]
                const formData = new FormData()
                formData.append('upload_preset', process.env.UPLOAD_PRESET)
                formData.append('file', profile_picture)
                setLoading(true) 
                setPhoto('')               
                axios.post(process.env.IMAGE_UPLOAD_LINK, formData)
                .then(res => {setFieldValue("profile_picture", res.data.secure_url); setLoading(false); setPhoto(res.data.secure_url)})
                .catch(err => { setLoading(false) })
              }}
              />
              
              <label htmlFor="file" className={classes.label}>
                <i className="fa fa-cloud-upload" aria-hidden="true"></i> 
                <div>
                {t("Upload Profile Image")} {loading ? <CircularProgress className={classes.uploadSpinner}/> : ''}
                </div>

              </label>
            </Box>
            <Field 
            name='bio' 
            label={t("Biography")}
            as={TextField} 
            margin='normal'
              inputprops={{
                startAdornment: (
                  <InputAdornment position="start">
                    <Subject />
                  </InputAdornment>
                ),
              }}
            />
            <Field 
            name='occupation' 
            label={t("Occupation")}
            as={TextField} 
            margin='normal'
              inputprops={{
                startAdornment: (
                  <InputAdornment position="start">
                    <EventBusy />
                  </InputAdornment>
                ),
              }}
            />
            <FormControl>
              <Field 
              name="address" 
              label={t("Address")}
              as={TextField} 
                inputprops={{
                  startAdornment: (
                    <InputAdornment position="start">
                    <EventBusy />
                  </InputAdornment>
                  ),
                }}
              />
            </FormControl>
            <FormControl>
              <InputLabel>{t("Language")}</InputLabel>
              <Field 
              as={Select} 
              name="language" 
                inputprops={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Language />
                    </InputAdornment>
                  ),
                }}
              >
                <MenuItem selected value="Kinyarwanda">{t("Kinyarwanda")}</MenuItem>
                <MenuItem value="English">{t("English")}</MenuItem>
                <MenuItem value="Franch">{t("French")}</MenuItem>
              </Field>
            </FormControl>
            
            <div>
              <Button
                type='submit'
                
                variant='contained'
                color='primary'
                className={classes.button}
                onClick={() => {
                  setDirection('back'); 
                }}
              >
                {t("Back")}
              </Button>
              <Button
                type='submit'
                id='backBtn'
                variant='contained'
                color='primary'
                className={classes.button}
                onClick={() => setDirection('forward')}
              >
                {t("Next")}
              </Button>
            </div>
          </Form>
        )}
      </Formik>
    </>
  );
};

SecondStep.propTypes = {
  formData: PropTypes.object.isRequired,
  setFormData: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired,
  prevStep: PropTypes.func.isRequired
};
