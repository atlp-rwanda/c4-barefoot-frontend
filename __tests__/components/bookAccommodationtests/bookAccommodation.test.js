import { shallow, mount } from "enzyme";
import React from 'react';
import toJson from "enzyme-to-json";
import { FirstStep } from "../../../src/components/bookAccommodation/firstStep";
import { SecondStep } from "../../../src/components/bookAccommodation/secondStep";
//import Signup from '../../src/components/signup'
import { ThirdStep } from "../../../src/components/bookAccommodation/thirdStep";
//import SocialButtons from '../../src/components/signup/socialButton'
//import SideDiv from "../../src/components/signup/sideDiv";
import {act} from 'react-dom/test-utils'
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { accommodationsPayload,bookAccommodationState,accommodationState} from '../../../dummyData'
import thunk from "redux-thunk";

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
let store;
describe('Book Accommodation', () => {
    let formData = new FormData()
    store = mockStore({
        fetchAccommodations:  accommodationState,
        bookAccommodations:bookAccommodationState
      });
    const props = {
        accommodation:{
      
            id: "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
            country: "Rwanda",
            city: "Kigali",
            title: "Marriot Hotel",
            description: "A serene environment for relaxation",
            photos: "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
        
        },
        selectedAccommodation:'1243534gfdggs-ggfdrdrd-t66fytftf',
        accommodations:accommodationsPayload.rows,
        count:6,
        snackBarMessage: {
            open: false,
            severity: '',
            message: null
          },
        formData:formData,
        clearBookSnackbar:jest.fn(),
        setFormData: function(){},
        nextStep: function(){},
        prevStep: function(){},
        profile_picture: 'image.jpg'
    }
    describe('Signup Rendering', ()=>{
        
        it('renders correctly first step', () => {
            const wrapper = shallow(<Provider store={store}><FirstStep {...props}/></Provider>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        // it('renders correctly Signup index', () => {
        //     const wrapper = shallow(<Signup {...props}/>)
        //     expect(toJson(wrapper)).toMatchSnapshot();
        // });
        it('renders correctly Third step', () => {
            const wrapper = shallow(<Provider store={store}><ThirdStep {...props}/></Provider>)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        // it('renders side div correctly', () => {
        //     const wrapper = shallow(<SideDiv />)
        //     expect(toJson(wrapper)).toMatchSnapshot();
        // });
        // it('renders Social buttons correctly', () => {
        //     const wrapper = shallow(<SocialButtons />)            
        //     expect(toJson(wrapper)).toMatchSnapshot();
        // });
        
    })
    describe('Functionalities', () => {
        // it('should call onChange prop with input value', () => {
        
        //         const onSearchMock = jest.fn();
        //         const component = mount(<Provider store={store}><SecondStep onChange={onSearchMock} {...props}/></Provider>);
        //         act( () => {   
        //             component.find('#file').simulate('change');
        //         })
            
        // });
        it('should call onSubmit on first step', () => {
        
            const onSubmit = jest.fn();
            const component = mount(<Provider store={store}><FirstStep onSubmit={onSubmit} {...props}/></Provider>);
            act( () => {   
                component.find('[type="submit"]').first().simulate('click');
            })
        
        });

        it('should call onSubmit on first step', () => {
        
            const handleChange = jest.fn();
            const component = mount(<Provider store={store}><FirstStep onChange={handleChange} {...props}/></Provider>);
            act( () => {   
                component.find('#pagination').first().simulate('change');
            })
        
        });

        it('should call onSubmit on Second step', () => {
        
            const onSubmit = jest.fn();
            const component = mount(<Provider store={store}><SecondStep onSubmit={onSubmit} {...props} values="custom value" /></Provider>);
            act( () => {   
                component.find('[type="submit"]').first().simulate('click');
            })
        
        });
        it('should call onSubmit on Third step', () => {
        
            const onClick = jest.fn();
            const component = mount(<Provider store={store}><ThirdStep onClick={onClick} {...props} values="custom value" /></Provider>);
            act( () => {   
                component.find('[id="backBtn"]').first().simulate('click');
            })
        
        });
        it('should call onClick for backward on second step', async() => {
            
            const signupRequest = jest.fn()
                         
            const component = mount(<Provider store={store}><SecondStep onClick={signupRequest} {...props} /></Provider>) 
            const button = component.find('#backBtn').at(0)
            await act( async() => {
                button.simulate('click');
            });
            expect(button.length).toBe(1)
            expect(signupRequest.mock.calls.length).toEqual(0);
        });
        it('should call onClick to submit form', async() => {
            
            const signupRequest = jest.fn()
                         
            const component = mount(<Provider store={store}><ThirdStep {...props} /></Provider>) 
            const button = component.find('[btn="submitBtn"]').first()
            await act( async() => {
                button.simulate('click');
            });
            expect(button.length).toBe(1)
            expect(signupRequest.mock.calls.length).toEqual(0);
        });
        // it('submit form', async() => {
        //     const component = mount(<Provider store={store}><FirstStep {...props} /></Provider>) 
        //     const button = component.find('[form-data="form-1"]').first()
        //     await act( async() => {
        //         button.simulate('submit');
        //     });
        //     expect(button.length).toBe(1)
        // });

    })

})
