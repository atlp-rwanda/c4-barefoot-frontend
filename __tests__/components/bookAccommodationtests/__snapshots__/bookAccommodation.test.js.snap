// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Accommodation Signup Rendering renders correctly Third step 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <ThirdStep
    accommodation={
      Object {
        "city": "Kigali",
        "country": "Rwanda",
        "description": "A serene environment for relaxation",
        "id": "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        "photos": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
        "title": "Marriot Hotel",
      }
    }
    accommodations={
      Array [
        Object {
          "city": "Kigali",
          "country": "Rwanda",
          "description": "A serene environment for relaxation",
          "id": "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
          "photos": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
          "title": "Marriot Hotel",
        },
      ]
    }
    clearBookSnackbar={[MockFunction]}
    count={6}
    formData={FormData {}}
    nextStep={[Function]}
    prevStep={[Function]}
    profile_picture="image.jpg"
    selectedAccommodation="1243534gfdggs-ggfdrdrd-t66fytftf"
    setFormData={[Function]}
    snackBarMessage={
      Object {
        "message": null,
        "open": false,
        "severity": "",
      }
    }
  />
</ContextProvider>
`;

exports[`Book Accommodation Signup Rendering renders correctly first step 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <FirstStep
    accommodation={
      Object {
        "city": "Kigali",
        "country": "Rwanda",
        "description": "A serene environment for relaxation",
        "id": "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
        "photos": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
        "title": "Marriot Hotel",
      }
    }
    accommodations={
      Array [
        Object {
          "city": "Kigali",
          "country": "Rwanda",
          "description": "A serene environment for relaxation",
          "id": "0ce36391-2c08-3074-bddb-a4ea8cccbbc5",
          "photos": "https://cache.marriott.com/marriottassets/marriott/KGLMC/kglmc-exterior-0030-hor-feat.jpg",
          "title": "Marriot Hotel",
        },
      ]
    }
    clearBookSnackbar={[MockFunction]}
    count={6}
    formData={FormData {}}
    nextStep={[Function]}
    prevStep={[Function]}
    profile_picture="image.jpg"
    selectedAccommodation="1243534gfdggs-ggfdrdrd-t66fytftf"
    setFormData={[Function]}
    snackBarMessage={
      Object {
        "message": null,
        "open": false,
        "severity": "",
      }
    }
  />
</ContextProvider>
`;
